### PR TITLE
[Windows] Fix copy to clipboard

### LINKED ISSUE #1107: Garbage in clipboard when copy non ASCII characters
**Steps to reproduce**
* open chat and copy some text with non ASCII characters
* paste it anywhere; at the end there would be some garbage: `��煱뺺䤷`

If that's not working try to copy different parts at the end of message.

tags: unicode, utf-8

**Env**
Win 7, 8.1 x64, uTox v.0.16.1

### PR DESCRIPTION
**Pull request by redmanmale**
Fixes #1107.

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/1132)

### CODE REVIEW

**Comment by endoffile78** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/1132#-:-KzyxiMo7MxvVbiZHnrM:bnfp4nl)*

**Comment by GrayHatter** [comment]
You left in the TODO comments?

How hard would it be to malloc here as needed?

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/1132#-:-L4gKySQniq0p3JyRst8:b-pv2m3b)*

**Comment by redmanmale** [comment]
I left them as they were before.

Maybe not that hard but I didn't really think about it.
This was supposed to be a quick small fix.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/1132#-:-L4kHS9vCHZU_70sFP5l:bup6v7z)*

File: src/windows/main.c [hunk 1]
```diff
@@ -291,8 +291,9 @@ void setselection(char *UNUSED(data), uint16_t UNUSED(length)) {
 }
 
 void copy(int value) {
-    const uint16_t max_size = INT16_MAX + 1;
+    const uint32_t max_size = UINT16_MAX + 1;
```
Comment (GrayHatter) [inline src/windows/main.c]: `UINT16_MAX + 1` that's a scary line of code to read... -_-

**Comment by GrayHatter** [comment]
I can't LGTM with comments that are out of date :/

**Comment by redmanmale** [comment]
These comments are about de-hardcoding buffers' sizes and it haven't done yet.

File: src/windows/main.c (hunk 1 above)
Comment (redmanmale) [reply src/windows/main.c]: I can rewrite this without such value but there will be some `-1`s instead.

**Comment by GrayHatter** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/main.c, line 294 at r1](https://reviewable.io:443/reviews/utox/utox/1132#-KzyanOP2jSNSRNDHPL3-r1-294:-L4wEJo4KqHr1w5HwN9Q:b-oj8t7q) ([raw file](https://github.com/utox/utox/blob/88175e4379fda750258be336a7c1c62296037c01/src/windows/main.c#L294)):*
<details><summary><i>Previously, redmanmale (redmanmale) wrote…</i></summary><blockquote>

I can rewrite this without such value but there will be some `-1`s instead.
</blockquote></details>

no, it's fine here, it's just something that should make any reviewer tense up a bit when reading it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/1132#-:-L4wETPUmxll_4QToikh:bnfp4nl)*

**Comment by GrayHatter** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/1132)*

**Pull request merged**